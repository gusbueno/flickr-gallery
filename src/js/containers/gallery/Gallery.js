'use strict'

import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import { requestPhotos } from '../../actions/gallery'
import { requestPhotoDetail } from '../../actions/photoDetail'
import styles from './Gallery.scss'
import PhotoItem from '../../components/photoItem/PhotoItem'

class Gallery extends PureComponent {
  componentDidMount () {
    this.props.onRequestPhotos('ireland') // default term: 'ireland'
  }

  _renderPhotos () {
    return this.props.photos.map(photo => {
      return <PhotoItem key={photo.id} photo={photo} onRequestPhotoDetail={this.props.onRequestPhotoDetail} />
    })
  }

  render () {
    return (
      <div className={styles.gallery}>
        <div className={styles['photo-list']}>
          {this._renderPhotos()}
        </div>
      </div>
    )
  }
}

const mapStateToProps = (state) => {
  return ({
    photos: state.gallery.photos
  })
}

const mapDispatchToProps = (dispatch) => {
  return ({
    onRequestPhotos: (term) => {
      dispatch(requestPhotos(term))
    },
    onRequestPhotoDetail: (photo) => {
      dispatch(requestPhotoDetail(photo))
    }
  })
}

Gallery.propTypes = {
  photos: PropTypes.array.isRequired,
  onRequestPhotos: PropTypes.func.isRequired
}

export default connect(mapStateToProps, mapDispatchToProps)(Gallery)
