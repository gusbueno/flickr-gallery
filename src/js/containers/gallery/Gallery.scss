.gallery {
  display: flex;

  .photo-list {
    display: flex;
    flex-wrap: wrap;
    flex-direction: row;
    flex: 1;
  }
}