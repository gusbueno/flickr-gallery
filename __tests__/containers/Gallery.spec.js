'use strict'

/* global describe, it, expect */

import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'

import Gallery from '../../src/js/containers/gallery/Gallery'

const storeFake = {
  default: () => {},
  subscribe: () => {},
  dispatch: () => {},
  getState: () => {
    return {
      gallery: {
        page: 1,
        pages: 200,
        photos: [
          {
            title: 'Title',
            farm: 123,
            server: '123',
            id: '123',
            secret: '123',
            owner: '123',
            ownername: 'Pepe'
          }
        ]
      }
    }
  }
}

describe('<Gallery /> container', () => {
  it('should match with the snapshot', () => {
    const tree = renderer.create(
      <Provider store={storeFake}>
        <Gallery />
      </Provider>
    )
    const json = tree.toJSON()
    expect(json).toMatchSnapshot()
  })

  const galleryShallow = shallow(
    <Provider store={storeFake}>
      <Gallery />
    </Provider>
  )

  it('should exists', () => {
    console.log(galleryShallow.instance())
    expect(galleryShallow.length).toEqual(1)
  })
})
