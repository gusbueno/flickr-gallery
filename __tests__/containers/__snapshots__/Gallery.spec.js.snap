// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Gallery /> container should match with the snapshot 1`] = `
<div
  className="gallery"
>
  <div
    className="pagination"
  >
    <div
      className="page-button current-page"
      onClick={[Function]}
    >
      1
    </div>
    <div
      className="page-button false"
      onClick={[Function]}
    >
      2
    </div>
    <div
      className="page-button false"
      onClick={[Function]}
    >
      3
    </div>
    <div
      className="page-button false"
      onClick={[Function]}
    >
      4
    </div>
    <div
      className="page-button false"
      onClick={[Function]}
    >
      5
    </div>
    <div
      className="page-button false"
      onClick={[Function]}
    >
      6
    </div>
    <div
      className="page-button false"
      onClick={[Function]}
    >
      7
    </div>
    <div
      className="page-button false"
      onClick={[Function]}
    >
      8
    </div>
    <div
      className="page-button false"
      onClick={[Function]}
    >
      9
    </div>
    <div
      className="page-button false"
      onClick={[Function]}
    >
      10
    </div>
    <div
      className="page-button"
      onClick={[Function]}
    >
      Next
    </div>
  </div>
  <div
    className="photo-list"
  >
    <div
      className="photo-item"
    >
      <div
        className="photo-wrapper"
        onClick={[Function]}
      >
        <img
          alt="Title"
          src="https://farm123.staticflickr.com/123/123_123.jpg"
        />
      </div>
      <div
        className="photo-info"
      >
        <span
          className="photo-title"
        >
          Title
        </span>
        <div
          className="photo-owner-wrapper"
        >
          <a
            className="photo-owner-text"
            href="https://www.flickr.com/photos/123/123"
            target="blank"
          >
            Pepe
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
